File: crates/trusty-agents/ui/src/lib/executionStyle.ts
// Presentation helpers for the coding-delegation style selector (#4353;
// spec DOC-62 `docs/specs/DOC-62-style-modes-coding-delegation.md`).
//
// Why: DOC-62 makes two things true at once that a naive selector gets wrong.
// First, §5.4: a caller-supplied style is a CEILING REQUEST the callee may
// raise but never lower, so the style a user picks is frequently not the style
// that runs — today ALWAYS not, since SM-9 raises `vibe` to `engineer` while
// #2596 is open and the coding lane's own floor raises `hack` to `vibe` first.
// A control that rendered its own selected value would therefore be stating
// something false; OQ-6's answer is to render the EFFECTIVE style and its
// resolution path instead. Second, OQ-3: the style value `engineer` collides
// with the tcode `engineer` SUB-AGENT role, and a user who reads
// `style = engineer` as "delegate to the engineer agent" has read it as
// something epic #4345 decision 2 forbids. DOC-62's recommendation is to keep
// the wire value and disambiguate in the GUI label — so the disambiguation
// lives here, in one place, rather than being re-typed at each render site
// where it could quietly be dropped. That recommendation is a RECOMMENDATION:
// OQ-3 is not formally ratified.
//
// What: pure lookups over the server-computed payload — labels, one-line
// meanings, and the human sentences for a resolution's source and escalations.
// NOTHING here resolves a style. Precedence, the §5.4 ceiling and the SM-9
// fail-safe are computed server-side by the real `ResolvedStyle::resolve` and
// arrive on `SubagentCoding.resolutions`; re-deriving any of it client-side
// would be the second copy of a gate that `agent_subagents.rs`'s honesty rules
// exist to forbid, and would silently go stale the day #2596 lands.
//
// Test: `executionStyle.test.ts`.

import type {
  ExecutionStyle,
  StyleResolution,
  StyleResolutionRow,
  SubagentCoding,
} from './agentConfig';

/** How one style is presented: a disambiguated label plus what it means. */
export interface StylePresentation {
  /** The GUI label. Never the bare wire value for `engineer` — see OQ-3. */
  label: string;
  /** One line on the ceremony level, from DOC-62 §1's table. */
  meaning: string;
}

/**
 * Why: DOC-62 §1 fixes the three names to the already-ratified Execution
 * Patterns tiers, and OQ-3's recommended mitigation for the `engineer`
 * collision is presentational — which makes it "easy to lose" (the spec's own
 * words). Keeping the mitigation in a single exported table is what stops it
 * being lost: a render site cannot accidentally print the raw wire value,
 * because the raw value is not what it is handed.
 * What: label + meaning per style. `engineer`'s label says "full loop" and its
 * meaning states explicitly that it is a ceremony level and not the tcode
 * `engineer` sub-agent, which is the whole point of the entry.
 * Test: `styleLabel_disambiguates_engineer_from_the_subagent_role`,
 * `stylePresentation_covers_every_style`.
 */
export const STYLE_PRESENTATION: Record<ExecutionStyle, StylePresentation> = {
  hack: {
    label: 'Hack — quick ops',
    meaning:
      'Trivial: a direct answer or a couple of reads, with no engineering ceremony at all.',
  },
  vibe: {
    label: 'Vibe — quick coding',
    meaning:
      'A small, self-contained change: no full test suite, no spec, no tickets. The tier itself is not implemented yet (#2596).',
  },
  engineer: {
    label: 'Engineer — full loop',
    meaning:
      'The full lifecycle: spec → issue → worktree → branch → PR → review → merge → CI. This is a CEREMONY LEVEL, not the tcode "engineer" sub-agent — coding work is only ever delegated to the coding PM.',
  },
};

/** The disambiguated GUI label for a style. */
export function styleLabel(style: ExecutionStyle): string {
  return STYLE_PRESENTATION[style].label;
}

/**
 * Why: the resolution path (DOC-62 §3.4) is only useful if a reader can tell
 * WHICH level supplied the value — "your `agent.toml` was read" and "nothing
 * supplied one, so the built-in applied" are different answers to the same
 * user question, and both look identical if only the value is shown.
 * What: a sentence fragment naming the precedence level.
 * Test: `sourceSentence_names_each_precedence_level`.
 */
export function sourceSentence(source: StyleResolution['source']): string {
  switch (source) {
    case 'caller':
      return 'requested on this delegation';
    case 'config':
      return 'from this agent’s [subagents] default_style';
    case 'built-in':
      return 'the built-in default (nothing supplied one)';
  }
}

/**
 * Why: DOC-62 §3.4 requires "an explicit statement when the effective style
 * differs from the requested style, with the reason". `callee-floor` and
 * `tier-unimplemented` are wire tags; a user needs the reason, and SM-9 in
 * particular must not read as a bug — `vibe` running `engineer` is the
 * specified, deliberate fail-safe, and saying so is what stops it being
 * reported as a defect.
 * What: one sentence per escalation reason. Unknown tags (a newer sidecar)
 * degrade to a neutral statement rather than being dropped: an unexplained
 * raise is still a raise the user is entitled to see.
 * Test: `escalationSentence_explains_the_two_specified_reasons`,
 * `escalationSentence_degrades_on_an_unknown_reason`.
 */
export function escalationSentence(reason: string): string {
  switch (reason) {
    case 'callee-floor':
      return 'The coding lane enforces a higher floor: a delegation addressed to the coding PM is by definition asking for a code change, so it can never run below that floor (DOC-62 §5.4).';
    case 'tier-unimplemented':
      return 'That tier has no implementation yet, so the next higher one ran and said so, rather than silently accepting the request and running less (DOC-62 SM-9, #2596).';
    default:
      return `Raised for an unrecognised reason reported by the server (${reason}).`;
  }
}

/**
 * Why: the selector's controls and the payload's rows must be matched by the
 * `caller` key, not by index — the server owns the row order and a positional
 * lookup would silently mis-attribute a resolution to the wrong control if that
 * order ever changed. A missing row is returned as `null` so the caller renders
 * "unreported" rather than a fabricated resolution.
 * What: the row whose `caller` equals `caller` (`null` = the no-override row).
 * Test: `resolutionFor_matches_by_caller_not_by_position`,
 * `resolutionFor_returns_null_for_an_unreported_row`.
 */
export function resolutionFor(
  coding: SubagentCoding | null | undefined,
  caller: ExecutionStyle | null,
): StyleResolution | null {
  const row: StyleResolutionRow | undefined = coding?.resolutions?.find(
    (r) => r.caller === caller,
  );
  return row?.resolution ?? null;
}

/**
 * Why: "did I get what I asked for?" is the one question the pane exists to
 * answer honestly, and it is NOT the same as "are there escalations" for the
 * no-override row — there, `requested` is whatever the config or built-in
 * supplied, and comparing against the user's (absent) choice would be
 * meaningless. Comparing `requested` against `effective` is the comparison
 * DOC-62 §3.4 actually specifies.
 * What: true when the resolved value differs from the one that entered
 * resolution.
 * Test: `wasOverruled_is_true_only_when_effective_differs_from_requested`.
 */
export function wasOverruled(resolution: StyleResolution): boolean {
  return resolution.requested !== null && resolution.requested !== resolution.effective;
}

File: crates/trusty-agents/src/tools/execution_style.rs
//! `ExecutionStyle` — the ceremony level a caller may attach to a coding
//! delegation, and the one-way resolution that turns a caller REQUEST into an
//! effective style (issues #4349/#4350; spec DOC-62 §5, in review as PR #4529).
//!
//! Why: a caller-supplied style is an **author tag**, and this repository's own
//! captured research
//! (`docs/research/quality-gates-agent-prs-article-2026-04.md:122`) rules that
//! risk classification "must be automatic … not from author tags, since an
//! agent has no self-awareness of how risky its own change is". The spec
//! reconciles that with the owner's decision 3 (callers MAY name a style) by
//! making the value a **ceiling request the callee may raise but never lower**.
//! That asymmetry is the whole safety story of the feature, so it is enforced
//! by CONSTRUCTION here rather than by discipline at the call sites — the same
//! shape `ProposalEnvelope::for_cross_product`
//! (`crate::tools::cross_product`) uses to make DOC-41 §5.5's propose-only rule
//! "structural rather than a convention a future edit could forget".
//! What: [`ExecutionStyle`] is the closed three-variant vocabulary, declared and
//! `Ord`-ered by ascending ceremony so `max` IS "raise, never lower".
//! [`ResolvedStyle`] is the resolution result; its ONLY constructor is
//! [`ResolvedStyle::resolve`], its fields are private, and it exposes no
//! mutator — there is no code path through this module that yields an effective
//! style below the floor it was resolved against. [`StyleSource`] records which
//! precedence level supplied the value and [`StyleEscalation`] why the effective
//! value differs from the request, so a caller is never silently overruled.
//! Style is ceremony ONLY: nothing here reads or writes a tool list, a
//! permission, or a command line (DOC-62 SM-11).
//! Test: `execution_style_tests` — the ceiling property over the full
//! request×floor cross-product, the `vibe` → `engineer` fail-safe, precedence,
//! and the bounded, task-independent policy block.

use serde::{Deserialize, Serialize};

/// The closed vocabulary of ceremony levels (DOC-62 §5.1, epic #4345 decision
/// 1).
///
/// Why: a closed enum rather than a string is how decision 1's "no new axis is
/// invented" becomes mechanical — an open field is a place for a fourth,
/// undocumented tier to appear without a decision. The three names are the
/// already-ratified Execution Patterns tiers
/// (`docs/trusty-code/vision-and-architecture-spec.md` §5.10, §10 D3), not a
/// new taxonomy. An unrecognized wire value is a `serde` error the bridge
/// returns to the caller, never a silent fallback to a default: silently
/// mapping an unknown style onto a default is indistinguishable, from the
/// caller's side, from the style having been honoured.
/// What: `Hack` (QUICK OPS — no code change at all), `Vibe` (VIBE — a small
/// change, reduced ceremony), `Engineer` (FULL LOOP — today's behaviour).
/// **Variant order is load-bearing**: the derived `Ord` is ascending ceremony,
/// so `a.max(b)` is "the higher ceremony of the two" and every escalation in
/// this module is a `max`. Reordering these variants silently inverts the
/// safety property; [`ceremony_order_is_ascending`] fails if anyone does.
/// Test: `ceremony_order_is_ascending`, `styles_round_trip_lowercase`,
/// `an_unknown_style_string_is_a_deserialization_error`.
#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum ExecutionStyle {
    /// QUICK OPS — trivial, answerable in a couple of reads. Per DOC-62 §4.1
    /// this is the ABSENCE of a code change, not a code change with the checks
    /// removed.
    Hack,
    /// VIBE — a small, self-contained change with reduced pre-PR ceremony.
    /// Unimplemented today (#2596); see [`ExecutionStyle::implemented`].
    Vibe,
    /// FULL LOOP — spec → issue → worktree → branch → PR → review → merge → CI.
    /// The built-in default and the fail-safe direction.
    Engineer,
}

impl ExecutionStyle {
    /// Every variant, in ascending ceremony order (#4353).
    ///
    /// Why: the GUI style selector (#4353) has to draw one control per style
    /// and ask the resolver what each would actually do. Spelling the three
    /// names again at that call site would put the closed vocabulary decision 1
    /// fixes into a second place, which is precisely what the enum exists to
    /// prevent — so the one list lives here and every consumer iterates it.
    /// What: `[Hack, Vibe, Engineer]`, in the same ascending-ceremony order the
    /// derived `Ord` gives. **A new variant MUST be added here too**; the
    /// ordering half of that obligation is pinned mechanically below, and the
    /// membership half by [`ExecutionStyle::as_str`]'s exhaustive match, which
    /// stops compiling until a new variant is named.
    /// Test: `all_is_sorted_by_ascending_ceremony`,
    /// `all_round_trips_through_the_wire_form`.
    pub const ALL: [Self; 3] = [Self::Hack, Self::Vibe, Self::Engineer];

    /// The built-in default when neither the caller nor config supplies one
    /// (DOC-62 §5.2).
    ///
    /// Why: `Engineer` is the most ceremony and matches today's behaviour
    /// exactly, so an absent style is byte-equivalent in behaviour to
    /// pre-#4349 dispatch. Defaulting in the other direction would silently
    /// reduce ceremony for every existing caller.
    /// What: [`ExecutionStyle::Engineer`].
    /// Test: `built_in_default_is_engineer`, `precedence_falls_through_to_built_in`.
    pub const BUILT_IN_DEFAULT: Self = Self::Engineer;

    /// The style whose pipeline actually EXISTS today, and why it differs
    /// (DOC-62 §7.3, SM-9).
    ///
    /// Why: VIBE is unimplemented (#2596 open; vision spec §5.10 marks it "Not
    /// implemented"). Silently accepting `vibe` and running less would ship the
    /// reduced tier without the ratification vision spec §10 D3 requires, so
    /// the fail-safe degrades UPWARD — more ceremony — and says so. When #2596
    /// lands, this one function changes and nothing else does: no interface
    /// change, no re-plumbing.
    /// What: `(Engineer, Some(TierUnimplemented))` for `Vibe`; the style itself
    /// with `None` otherwise. Never returns a style below its input — the
    /// ceiling property holds through this step too.
    /// Test: `vibe_degrades_upward_to_engineer_and_says_why`,
    /// `implemented_never_lowers_ceremony`.
    pub fn implemented(self) -> (Self, Option<StyleEscalation>) {
        match self {
            Self::Vibe => (Self::Engineer, Some(StyleEscalation::TierUnimplemented)),
            other => (other, None),
        }
    }

    /// The lowercase wire/display name.
    ///
    /// Why: the policy preamble is plain text read by an LLM persona, so the
    /// rendered name must come from one place rather than being re-spelled at
    /// each format site (where it could drift from the serde representation).
    /// What: `"hack"` / `"vibe"` / `"engineer"`, matching `rename_all`.
    /// Test: `styles_round_trip_lowercase`.
    pub fn as_str(self) -> &'static str {
        match self {
            Self::Hack => "hack",
            Self::Vibe => "vibe",
            Self::Engineer => "engineer",
        }
    }
}

/// Which precedence level supplied the style (DOC-62 §5.3).
///
/// Why: §3.4's reporting contract entitles a caller to see WHICH level its
/// effective style came from, so an override that was never read is visible
/// rather than mysterious.
/// What: `Caller` (per-delegation parameter) > `Config` (per-agent default) >
/// `BuiltIn` ([`ExecutionStyle::BUILT_IN_DEFAULT`]), first-match-wins.
/// Test: `precedence_is_caller_then_config_then_built_in`.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize)]
#[serde(rename_all = "kebab-case")]
pub enum StyleSource {
    /// Supplied on the delegation call itself.
    Caller,
    /// Supplied by the calling agent's configuration default.
    Config,
    /// Neither of the above — [`ExecutionStyle::BUILT_IN_DEFAULT`].
    BuiltIn,
}

/// Why an effective style is HIGHER than the one that was requested.
///
/// Why: DOC-62 §3.4 requires "an explicit statement when the effective style
/// differs from the requested style, with the reason". Without it, a caller
/// cannot distinguish "you got what you asked for" from "we quietly ran
/// something else", which is the failure mode SM-4 exists to prevent.
/// What: the two reasons this module can produce. Both are raises; there is no
/// variant for a reduction because no code path produces one.
/// Test: `vibe_degrades_upward_to_engineer_and_says_why`,
/// `a_floor_above_the_request_is_reported_as_an_escalation`.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize)]
#[serde(rename_all = "kebab-case")]
pub enum StyleEscalation {
    /// The callee's own lane enforces a higher floor than the request
    /// (DOC-62 §5.4).
    CalleeFloor,
    /// The requested tier has no implementation yet, so the next higher one
    /// ran (DOC-62 §7.3, SM-9; #2596).
    TierUnimplemented,
}

impl StyleEscalation {
    /// Human-readable tag used in the policy preamble and in reports.
    ///
    /// Why: same single-source-of-truth reason as [`ExecutionStyle::as_str`];
    /// SM-9 names the wire reason `tier-unimplemented` verbatim, so the string
    /// must not be re-spelled at render sites.
    /// What: the kebab-case serde name.
    /// Test: `escalation_tags_match_the_spec_wire_names`.
    pub fn as_str(self) -> &'static str {
        match self {
            Self::CalleeFloor => "callee-floor",
            Self::TierUnimplemented => "tier-unimplemented",
        }
    }
}

/// A resolved style: what was asked for, what will actually run, and why they
/// differ (DOC-62 §5.3, §5.4, §3.4).
///
/// Why: **the ceiling property is structural here, not conventional.** Fields
/// are private, there is no `Default`, no public constructor other than
/// [`ResolvedStyle::resolve`], and no mutator; `resolve` computes `effective`
/// only through `max` and [`ExecutionStyle::implemented`], both of which are
/// monotonically non-decreasing in ceremony. There is therefore no code path
/// through this type that yields an effective style below the floor it was
/// resolved against, regardless of what the caller requested — exactly the
/// guarantee `ProposalEnvelope::for_cross_product` gives `Disposition`.
/// Note what this type deliberately does NOT have: any accessor a caller could
/// use to reach a tool list, a permission, or a command line. Style is a
/// ceremony record and nothing else (DOC-62 SM-11).
/// What: `requested` (the value that entered resolution, `None` when nothing
/// was supplied at any level), `source`, `effective`, and `escalations` (empty
/// when the request was honoured as-is).
/// Test: `a_caller_can_never_lower_ceremony_below_the_callee_floor`,
/// `resolution_is_reported_end_to_end`, `precedence_is_caller_then_config_then_built_in`.
#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
pub struct ResolvedStyle {
    /// The style that entered resolution, or `None` when no level supplied one.
    requested: Option<ExecutionStyle>,
    /// Which precedence level [`Self::requested`] came from.
    source: StyleSource,
    /// The style that will actually be applied — never below the floor.
    effective: ExecutionStyle,
    /// Why `effective` differs from `requested`; empty when it does not.
    escalations: Vec<StyleEscalation>,
}

impl ResolvedStyle {
    /// Resolve caller > config > built-in, then raise to the callee's floor and
    /// to the nearest implemented tier.
    ///
    /// Why: the ONLY constructor, deliberately (see the type docs). Every input
    /// is combined with `max`, so the function is monotonically non-decreasing
    /// in ceremony: a caller asking for LESS than `callee_floor` moves the
    /// effective value up to the floor and is told so, and can never move it
    /// down. That is DOC-62 §5.4 — "the tcode PM MAY apply more ceremony than
    /// the resolved style requests; it MUST NOT apply less" — made mechanical.
    /// What: first-match-wins over `caller` then `config` then
    /// [`ExecutionStyle::BUILT_IN_DEFAULT`]; the result is `max`-ed with
    /// `callee_floor` (recording [`StyleEscalation::CalleeFloor`] if that
    /// raised it) and then passed through [`ExecutionStyle::implemented`]
    /// (recording [`StyleEscalation::TierUnimplemented`] if that raised it
    /// again). Both escalations can apply to one resolution and both are
    /// reported, in the order they were applied.
    /// Test: `a_caller_can_never_lower_ceremony_below_the_callee_floor`,
    /// `precedence_is_caller_then_config_then_built_in`,
    /// `vibe_degrades_upward_to_engineer_and_says_why`,
    /// `both_escalations_are_reported_when_both_apply`.
    pub fn resolve(
        caller: Option<ExecutionStyle>,
        config: Option<ExecutionStyle>,
        callee_floor: ExecutionStyle,
    ) -> Self {
        let (requested, source) = match (caller, config) {
            (Some(style), _) => (Some(style), StyleSource::Caller),
            (None, Some(style)) => (Some(style), StyleSource::Config),
            (None, None) => (None, StyleSource::BuiltIn),
        };
        let asked = requested.unwrap_or(ExecutionStyle::BUILT_IN_DEFAULT);

        let mut escalations = Vec::new();
        // Raise (never lower) to the lane's floor.
        let floored = asked.max(callee_floor);
        if floored > asked {
            escalations.push(StyleEscalation::CalleeFloor);
        }
        // Raise (never lower) to the nearest tier that actually exists.
        let (effective, tier) = floored.implemented();
        if let Some(reason) = tier {
            escalations.push(reason);
        }

        // Postcondition — the ceiling property, asserted at the one site that
        // can produce a `ResolvedStyle`. Cheap and debug-only; the property is
        // pinned unconditionally by
        // `a_caller_can_never_lower_ceremony_below_the_callee_floor`.
        debug_assert!(effective >= callee_floor, "resolution lowered below floor");
        debug_assert!(effective >= asked, "resolution lowered below the request");

        Self {
            requested,
            source,
            effective,
            escalations,
        }
    }

    /// The style that will actually be applied.
    ///
    /// Why: callers that report or branch on style must read the EFFECTIVE
    /// value, never the request — reading the request is how "not run" quietly
    /// becomes "fine" (DOC-62 SM-4).
    /// What: the resolved value, always `>=` both the request and the floor.
    /// Test: `resolution_is_reported_end_to_end`.
    pub fn effective(&self) -> ExecutionStyle {
        self.effective
    }

    /// Which precedence level supplied the requested value.
    ///
    /// Why: DOC-62 §3.4's reporting contract (AC-4).
    /// What: see [`StyleSource`].
    /// Test: `precedence_is_caller_then_config_then_built_in`.
    pub fn source(&self) -> StyleSource {
        self.source
    }

    /// Why the effective style differs from the request; empty when it does not.
    ///
    /// Why: DOC-62 §3.4 — a caller that asked for `vibe` is entitled to know
    /// exactly what it bought.
    /// What: raises only, in application order.
    /// Test: `both_escalations_are_reported_when_both_apply`.
    pub fn escalations(&self) -> &[StyleEscalation] {
        &self.escalations
    }

    /// Whether a style was supplied at all, rather than falling through to the
    /// built-in default.
    ///
    /// Why: DOC-62 §6.4 requires that an absent style render NO policy block,
    /// so the no-style path stays byte-identical to pre-#4349 dispatch. This is
    /// the predicate that keeps that promise, and it deliberately asks about
    /// the SOURCE rather than the effective value (which is always populated).
    /// What: true iff [`Self::source`] is not [`StyleSource::BuiltIn`].
    /// Test: `an_unrequested_style_renders_no_policy_block`.
    pub fn is_explicit(&self) -> bool {
        !matches!(self.source, StyleSource::BuiltIn)
    }

    /// Render the TA-PM-policy block appended to a cross-product preamble
    /// (#4349; DOC-62 §6.1, SM-7).
    ///
    /// Why: the product boundary is one argv slot — `tcode run-task <agent>
    /// <task> --project <dir> --json` — so a structured handoff can only cross
    /// it flattened into the task string (see
    /// `HandoffContext::render_preamble`'s docs). SM-7 fixes what this text MAY
    /// carry: the effective style, its meaning, decision 2's PM-only statement
    /// as INFORMATION, and the §3 gate boundary as INSTRUCTION. SM-8 fixes what
    /// it may not be: the sole enforcement of any security control. Both rules
    /// are visible in the text itself — the `NON_CODING_TARGETS` line says out
    /// loud that it is describing a code-enforced property, so no future reader
    /// mistakes the paragraph for the mechanism.
    /// What: a short block whose length is bounded and INDEPENDENT of the task
    /// text or of any caller-supplied field, so the 4 KiB handoff cap never
    /// becomes a function of task size (DOC-62 §6.4). Rendered after the
    /// caller's own lines and under a heading that labels it as not
    /// caller-supplied.
    /// Test: `policy_block_states_effective_style_and_the_gate_boundary`,
    /// `policy_block_is_bounded_and_task_independent`,
    /// `policy_block_reports_the_tier_unimplemented_fallback`.
    pub fn render_policy_block(&self) -> String {
        let mut out = String::from("Delegation policy (system-supplied, not from the caller):\n");
        out.push_str(&format!(
            "- Effective execution style: {}.\n",
            self.effective.as_str()
        ));
        if let Some(requested) = self.requested.filter(|r| *r != self.effective) {
            let reasons: Vec<&str> = self.escalations.iter().map(|e| e.as_str()).collect();
            out.push_str(&format!(
                "- Requested style was {}; raised to {} ({}). Ceremony may be raised, never lowered.\n",
                requested.as_str(),
                self.effective.as_str(),
                reasons.join(", ")
            ));
        }
        out.push_str(
            "- Style selects only your own internal ceremony. It never relaxes a check the \
             target repository enforces (CI, branch protection, required review), and is never \
             a reason to skip a review gate.\n\
             - A check that was not run is reported as not run, never as passed.\n\
             - Style grants no capability: it does not change which tools you may call or what \
             they may do.\n\
             - Coding work is delegated to you as the project manager; no coding sub-agent is \
             reachable from the caller. That boundary is enforced in code, not by this text.\n",
        );
        out
    }
}

#[cfg(test)]
#[path = "execution_style_tests.rs"]
mod execution_style_tests;
